### ISSUE TITLE
A question about feature extraction.

### ISSUE THREAD

**Issue by huangf79** [gallilmaimon/DISSC#3]
Thanks for sharing this work! I wonder is the code in [encode.py](https://github.com/gallilmaimon/DISSC/blob/main/data/encode.py) extracts and clusters representations from **the final layer** of HuBERT by default?

**Comment by gallilmaimon** [comment]
Hey, I appreciate your interest in our work :)

In short, it uses by default the **6th layer** of HuBERT with the pre-trained kmeans quantiser.

In a bit more details, the encode.py files uses SpeechEncoder from textlesslib - https://github.com/facebookresearch/textlesslib/blob/ba33d669d8284b4f7bfe81e7384e83ab799fe384/textless/data/speech_encoder.py#L55
And as you can see from - https://github.com/facebookresearch/textlesslib/blob/ba33d669d8284b4f7bfe81e7384e83ab799fe384/textless/__init__.py#L15 the default when using ```--model_name hubert-base-ls960``` is to take layer 6. You can change the model name to ```hubert-base-ls960-layer-9``` in order to take layer 9. We didn't really play around with this and just took the default. Maybe you can improve the results a bit more by optimizing this choice more (as well as fitting your own quantiser on your data).

**Comment by huangf79** [comment]
So it is a small mistake that you said 'We extracted representations from **the final layer** of HuBERT and set K = 100.' in the paper (section 3.1) ?

**Comment by gallilmaimon** [comment]
There is indeed a small discrepancy between this version in the code and what is mentioned there. In the end we used the 6th layer because preliminary results showed this was slightly better. We believe that this won't make a big difference but could be worth further investigation :)

**Comment by huangf79** [comment]
I get it. Thanks a lot!

**Comment by huangf79** [comment]
Hey. Sorry to reopen this issue, I tried to train a LenPredictor but got bad results. Even after 200 epochs, the LenSumLoss was still more than 1800 and the predicted durations were unreasonable.
(1) I wonder what the normal loss value is, also that for PitchPredictor. (I used just part of your code, maybe it's caused by my wrong implementation) 
(2) BTW, does the mean and var of the dataset matters? I set them 0 and 1 for convenience.

**Comment by gallilmaimon** [comment]
Hey again, generally speaking I prefer if you open a new issue for each separate thing so that people in the future can find it more easily :)

About your questions:
(1) a normal loss value depends on the distribution of the dataset (specifically the duration distribution), a general rule of thumb is that it should be a noticeably better loss than simply guessing the average length or the common value or something. Are you using one of the described datasets or a custom one? I would look at the distribution of gt length to see if maybe there is bug or maybe the Hubert model doesn't output good tokens for your dataset. Anyway your loss sounds like some sort of bug.
(2) the mean and var of which dataset? Of the durations? This can impact the absolute loss scale which would impact learning rate and also the relative weight to big errors (because of the MSE). This won't necessarily cause a big difference but hard to guarantee.
